config/config.go

```
@@ -22,36 +22,42 @@ type Config struct {
+	v.SetDefault("server_port", "8080")
+	v.SetDefault("log_level", "info")
+	v.SetDefault("gin_mode", "debug")
+	v.SetDefault("retry.base_delay", 100)
+	v.SetDefault("retry.max_delay", 5000)
+	v.SetDefault("retry.max_retries", 5)
+
+	// Tell Viper to automatically override values from environment variables
+	v.AutomaticEnv()
+
+	// Optional: If you want to keep your existing env var names
+	v.BindEnv("database_url", "DATABASE_URL")
```

golangci-lint

Error return value of `v.BindEnv` is not checked

(errcheck)